### PR TITLE
[Indicators Searcher] Fix IndicatorsSearcher Stop Condition

### PR DESCRIPTION
**Pull request by DeanArbel**
## Status
- [x] Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/42855
fixes: https://github.com/demisto/etc/issues/42925
fixes: https://github.com/demisto/etc/issues/42782

## Description
Fixed an issue where different integrations could get stuck in a search loop when searching for more indicators than are available in the system.

## Does it break backward compatibility?
   - [x] No

### CODE REVIEW

**Comment by glicht** [comment]
@DeanArbel please schedule a code review session

**Review by glicht — CHANGES_REQUESTED** [review]
Let's do this one TDD. Add a test that fails and then do the fix.

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 1]
```diff
@@ -8022,33 +8013,31 @@ def total(self):
...
     @limit.setter
     def limit(self, value):
-        self._next_limit = self._original_limit = value
+        self._limit = value
 
-    def _is_search_done(self):
+    def is_search_done(self):
         """
         Checks one of these conditions:
         1. self.limit is set, and it's updated to be less or equal to zero
```
Comment (tkatzir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8025]:
```suggestion
        1. self.limit is set, and it's updated to be less or equal to zero - return True
```

Mention the return value in each case.

**Review by tkatzir — CHANGES_REQUESTED** [review]

**Review by tkatzir — APPROVED** [review]

**Comment by DeanArbel** [comment]
Validations are failing because of:
* Workday_IAM - missing test playbook.
* GetIndicatorsByQuery - missing test playbook.

Gitlab build is failing because of:
* `TAXII Feed` is unstable (flaky results that are caused by HTTP connection issues)
* In some builds the pack installation is failing, causing the build to fail.

File: Packs/Base/ReleaseNotes/1_15_1.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
 
 #### Scripts
 ##### CommonServerPython
-- Maintenance and stability enhancements.
\ No newline at end of file
+- Maintenance and stability enhancements.
```
Comment (glicht) [inline Packs/Base/ReleaseNotes/1_15_1.md:4]: Why this change?

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 3]
```diff
@@ -131,17 +131,13 @@ def create_new_edl(request_args: RequestArguments) -> str:
         size=PAGE_SIZE,
         limit=limit
     )
-    iocs = []
+    iocs: List[dict] = []
     formatted_iocs: set = set()
-    while True:
-        current_limit = limit + (limit - len(formatted_iocs))
-        indicator_searcher.limit = current_limit
+    while not indicator_searcher.is_search_done():
```
Comment (DeanArbel) [inline Packs/EDL/Integrations/EDL/EDL.py:136]: Put a hard limit here - no more than 10 loops

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 4]
```diff
@@ -7985,37 +7985,29 @@ def __init__(self,
...
-        size = min(self._size, self._next_limit or self._size)
         res = self.search_indicators_by_version(from_date=self._from_date,
                                                 query=self._query,
-                                                size=size,
+                                                size=self._size,
                                                 to_date=self._to_date,
                                                 value=self._value)
         fetched_len = len(res.get('iocs') or [])
         if fetched_len == 0:
+            self.limit = 0
```
Comment (DeanArbel) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8007]: replace limit with count - i.e. instead of lowering the limit, we'll increase the count.

File: Packs/Base/ReleaseNotes/1_15_1.md (hunk 2 above)
Comment (DeanArbel) [reply Packs/Base/ReleaseNotes/1_15_1.md:4]: It's an inner change of the way the `IndicatorsSearcher` class works, but I'll change it to be something more infomercial.

Comment (DeanArbel) [reply Packs/Base/ReleaseNotes/1_15_1.md:4]: probably got picked up when merging from master. Reverting

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 5]
```diff
@@ -8028,33 +8016,34 @@ def total(self):
...
-        if self._search_is_done:
-            return True
-
-        reached_limit = isinstance(self._next_limit, int) and self._next_limit <= 0
+        reached_limit = self.limit is not None and self.limit <= self._total_iocs_fetched
         if reached_limit:
+            demisto.debug("IndicatorsSearcher has reached its limit: {}".format(self.limit))
+            # update limit to match _total_iocs_fetched value
+            if self._total_iocs_fetched > self.limit:
+                self.limit = self._total_iocs_fetched
```
Comment (DeanArbel) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:8037]: @glicht Please notice this line.
We update the `self.limit` with `self._total_iocs_fetched` to handle cases where it's greater than the limit - this will allow updates to limit to affect further searches that might be done while updating limit (otherwise `limit` can be lower than `_total_iocs_fetched` even after increasing its value by the calling integration).

**Review by glicht — APPROVED** [review]
